src/install_pypi/plan/test/mod.rs

```diff
@@ -644,10 +641,34 @@ fn test_archive_is_path() {
     );
     let plan = harness::install_planner_with_lock_dir(tmp.path().to_path_buf());
     let installs = plan
-        .plan_install(&site_packages, AllCached, &required.to_borrowed())
+        .plan(&site_packages, AllCached, &required.to_borrowed())
         .expect("should install");
     // Should not install package
     assert!(installs.reinstalls.is_empty());
     assert!(installs.local.is_empty());
     assert!(installs.remote.is_empty());
 }
+
+#[test]
+fn duplicates_are_not_extraneous() {
+    let site_packages = MockedSitePackages::new()
+        // Our managed package
+        .add_registry("aiofiles", "0.6.0", InstalledDistOptions::default())
+        // Package not managed by us
+        .add_registry(
+            "aiofiles",
+            "0.6.1",
+            InstalledDistOptions::default().with_installer("not-me"),
+        );
+
+    // We don't need any package
+    let required = RequiredPackages::new();
+
+    let plan = harness::install_planner();
+    let installs = plan
+        .plan(&site_packages, NoCache, &required.to_borrowed())
+        .expect("should install");
+
+    assert!(installs.extraneous.is_empty());
+    assert_eq!(installs.duplicates.len(), 1);
+}
```

Comment:
This test does kind of the thing that I described a package that is double in the prefix and owned by different installers should be marked duplicate.